awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,168 @@
+            num_credentials=Count('credentials', distinct=True)):  
+        counts[cred_type.id] = {'name': cred_type.name,
+                                'credential_count': cred_type.num_credentials
+                                }
+    return counts
+    
+    
+@register('inventory_counts')
+def inventory_counts(since):
+    counts = {}
+    from django.db.models import Count
+    for inv in models.Inventory.objects.annotate(
+            num_sources=Count('inventory_sources', distinct=True), 
+            num_hosts=Count('hosts', distinct=True)).only('id', 'name', 'kind'):
+        counts[inv.id] = {'name': inv.name,
+                          'kind': inv.kind,
+                          'hosts': inv.num_hosts,
+                          'sources': inv.num_sources
+                          }
+    return counts
+
+
+@register('projects_by_scm_type')
+def projects_by_scm_type(since):
+    counts = dict(
+        (t[0] or 'manual', 0)
+        for t in models.Project.SCM_TYPE_CHOICES
+    )
+    for result in models.Project.objects.values('scm_type').annotate(
+        count=Count('scm_type')
+    ).order_by('scm_type'):
+        counts[result['scm_type'] or 'manual'] = result['count']
+    return counts
+
+
+@register('job_counts')   # TODO: evaluate if we want this (was not an ask)  Also, may think about annotating rather than grabbing objects for efficiency 
+def job_counts(since):    # TODO: Optimize -- for example, all of these are going to need to be restrained to the last 24 hours/INSIGHTS_SCHEDULE
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
+    counts['successful_jobs'] = models.UnifiedJob.objects.filter(status='successful').count()
+    counts['cancelled_jobs'] = models.UnifiedJob.objects.filter(status='canceled').count()
+    counts['failed_jobs'] = models.UnifiedJob.objects.filter(status='failed').count()               
+    counts['error_jobs'] = models.UnifiedJob.objects.filter(status='error').count()   # Do we also want to include error, new, pending, waiting, running?
+    counts['new_jobs'] = models.UnifiedJob.objects.filter(status='new').count()       # Also, how much of this do we want `per instance`
+    counts['pending_jobs'] = models.UnifiedJob.objects.filter(status='pending').count()
+    counts['waiting_jobs'] = models.UnifiedJob.objects.filter(status='waiting').count()
+    counts['running_jobs'] = models.UnifiedJob.objects.filter(status='running').count()
+    for instance in models.Instance.objects.all():
+        counts[instance.id] = {'uuid': instance.uuid,
+                               'jobs_total': instance.jobs_total,       # this is _all_ jobs run by that node
+                               'jobs_running': instance.jobs_running,   # this is jobs in running & waiting state
+                               'launch_type': {'manual': models.UnifiedJob.objects.filter(launch_type='manual').count(), # I can definitely condense this
```

Comment:
These queries don't seem to be actually *restricted* the instance group in question - I assume we want these to be the jobs that _ran on_ these instances?